ambry-tools/src/main/java/com.github.ambry.tools/admin/DumpData.java

```diff
@@ -731,6 +765,10 @@ public int compare(Pair<Long, Long> o1, Pair<Long, Long> o2) {
       });
     }
 
+    /**
+     * Adds and merges the interval if need be
+     * @param newInterval the new interval that needs to be added
+     */
```

Comment:
In (maybe unlikely) case of overlapping intervals, is `MergedInterval` class still able to handle them correctly? Say we have two intervals t1-t2, and t3-t4, and t3>t1, t4<t2, t3<t4(of course). What would be `logUnCoveredRanges` show?

Not sure if there would be an easy fix, just modify the comparator method, so that if there are overlapping intervals, fail the test.